### PR TITLE
Feature/Store type flagged db providers

### PR DESCRIPTION
**Pull request by jan-schutte**

### CODE REVIEW

File: src/Nox.Types.EntityFramework/Enums/NoxDataStoreTypeFlags.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace Nox.Types.EntityFramework.Enums;
+
+[Flags]
+public enum NoxDataStoreTypeFlags
+{
+    EntityStore,        //Persists domain entities as defined in the solution.nox.yaml                             --
```
Comment (rochar) [inline src/Nox.Types.EntityFramework/Enums/NoxDataStoreTypeFlags.cs]: you could add the comment as Summary... so devs will see it more easily

File: src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs [hunk 2]
```diff
@@ -4,18 +4,20 @@
...
 using Nox.Types.EntityFramework.Configurations;
 using Nox.Types.EntityFramework.EntityBuilderAdapter;
+using Nox.Types.EntityFramework.Enums;
 using Npgsql;
 
 namespace Nox.EntityFramework.Postgres;
 
 public class PostgresDatabaseProvider: NoxDatabaseConfigurator, INoxDatabaseProvider 
 {
+    public string ConnectionString { get; protected set; } = string.Empty;
```
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs:14]: if this is for all providers? you can add them to the base class NoxDatabaseConfigurator

File: src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs [hunk 3]
```diff
@@ -54,4 +56,14 @@ public string ToTableNameForSqlRaw(string table, string schema)
     {
         throw new NotImplementedException();
     }
+
```
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs:59]: if this is for all providers? you can add them to the base class NoxDatabaseConfigurator

File: src/Nox.Types.EntityFramework/Enums/NoxDataStoreTypeFlags.cs (hunk 1 above)
Comment (jan-schutte) [reply src/Nox.Types.EntityFramework/Enums/NoxDataStoreTypeFlags.cs]: added to summary

File: src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs (hunk 3 above)
Comment (jan-schutte) [reply src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs:59]: Rather keep it on the provider, not all database provider will implement the configurator

File: src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs (hunk 2 above)
Comment (jan-schutte) [reply src/Nox.EntityFramework.Postgres/PostgresDatabaseProvider.cs:14]: Rather keep it on the provider, not all database provider will implement the configurator

**Review by rochar — APPROVED** [review]

**Pull request merged**